### PR TITLE
Camera IO update

### PR DESCRIPTION
**Pull request by braden6521**
- Fixed ImageAcquisition class to work with new generation of Basler cameras.
- Updated documentation in existing examples
- Created new examples.

### CODE REVIEW

**Review by e10harvey — CHANGES_REQUESTED** [review]
@braden6521: Please add some documentation for the hard coded values. Also, where are the tests for ImAgeAcquisition_DCAM_mono?

File: opencsp/common/lib/camera/ImageAcquisition_DCAM_mono.py [hunk 1]
```diff
@@ -104,7 +110,9 @@ def instance_matches(self, possible_matches: list[ImageAcquisitionAbstract]) ->
     def get_frame(self) -> np.ndarray:
         # Start frame capture
         self.cap.StartGrabbingMax(1)
-        grabResult = self.cap.RetrieveResult(5000, pylon.TimeoutHandling_ThrowException)
+        grabResult = self.cap.RetrieveResult(
+            int(self.cap.ExposureTimeRaw.Max / 1000 * 1.5), pylon.TimeoutHandling_ThrowException
```
[the remark this answers is not in the document]
Comment (braden6521) [reply opencsp/common/lib/camera/ImageAcquisition_DCAM_mono.py:114]: Good call, it's documented now.

**Comment by braden6521** [comment]
@e10harvey, this is another tricky case in terms of unit testing we don't have a good solution for. To test this piece of code, we need a camera and network card physically connected to the computer. In fact, we have no tests for: ImageAcquisition_DCAM_color, ImageAcquisition_DCAM_mono, or ImageAcquisition_MSMF because these all require hardware present. These classes (in addition to all GUIs) are not tested currently. 

Our solution up to this point has been to have a test for ImageAcquisitionAbstract (in ocmmon/lib/camera/test). Any suggestions are welcome.

**Comment by e10harvey** [comment]
> @e10harvey, this is another tricky case in terms of unit testing we don't have a good solution for. To test this piece of code, we need a camera and network card physically connected to the computer. In fact, we have no tests for: ImageAcquisition_DCAM_color, ImageAcquisition_DCAM_mono, or ImageAcquisition_MSMF because these all require hardware present. These classes (in addition to all GUIs) are not tested currently.
> 
> Our solution up to this point has been to have a test for ImageAcquisitionAbstract (in ocmmon/lib/camera/test). Any suggestions are welcome.

Ok, let's get this merged. We can setup a machine with the physical hardware needed to test this. I filed: https://github.com/sandialabs/OpenCSP/issues/174

**Review by e10harvey — APPROVED** [review]

**Pull request merged**